src/components/storage/CockpitStorageIntegration.jsx

```diff
@@ -247,6 +251,52 @@ const CheckStorageDialog = ({
         }
     }, [useConfiguredStorage, checkStep]);
 
+    useEffect(() => {
+        if (checkStep !== "luks") {
+            return;
+        }
+
+        const cockpitDevices = (
+            Object.keys(newMountPoints)
+                    .map(devicePath => ({
+                        devicePath,
+                        deviceName: getDeviceNameByPath(deviceData, devicePath),
+                    }))
+        );
+
+        const devicesToUnlock = (
+            cockpitDevices
+                    .filter(({ devicePath, deviceName }) => {
+                        return (
+                            newMountPoints[devicePath].type === "crypto" &&
+                            deviceData[deviceName].formatData.attrs.v.has_key !== "True"
+                        );
+                    })
+                    .map(({ devicePath, deviceName }) => ({
+                        deviceName,
+                        passphrase: cockpitPassphrases[devicePath],
+                    }))
+        );
+
+        if (devicesToUnlock.some(dev => !dev.passphrase)) {
+            onCritFail()({ message: _("Cockpit storage did not provide the passphrase to unlock encrypted device.") });
```

Comment:
Wouldn't Anaconda need to prompt the user for the passphrase in this case? It is always possible that some LUKS device is closed for some reason.

Reply:
I agree, that this can be improved. However, since that more of an edge condition, I would suggest to implement the unlock code plus test for it as followup.

Reply:
I just hacked it to be like this and that works fine:

```
    useEffect(() => {
        if (checkStep !== "luks") {
            return;
        }

        const devicesToUnlock = (
            Object.keys(cockpitPassphrases).map(dev => ({ deviceName: getDeviceNameByPath(deviceData, dev),
                                                          passphrase: cockpitPassphrases[dev]
                                                        })));

        console.log("devicesToUnlock", devicesToUnlock);

        Promise.all(devicesToUnlock.map(unlockDevice))
                .catch(exc => {
                    setCheckStep();
                    setError(exc);
                })
                .then(() => {
                    setCheckStep("prepare-partitioning");
                });
    }, [dispatch, checkStep, cockpitPassphrases, newMountPoints, deviceData, onCritFail, setError]);
```